cirq-core/cirq/ops/control_values_test.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from cirq.ops import control_values as cv
+
+
+def test_init_control_values():
+    tests = [
+        ([], [()]),
+        ([1], [(1,)]),
+        ([[0, 1], 1], [(0, 1), (1, 1)]),
+        ([[[0, 1], [1, 0]]], [(0, 1), (1, 0)]),
+    ]
+    for control_values, want in tests:
+        control_vals = cv.ControlValues(control_values)
+        got = [cv.flatten(product) for product in control_vals]
+        assert want == sorted(got)
+
```

Comment:
we have a helper for testing equality.  could you and an EqualsTester test?

Reply:
done